### PR TITLE
feat(StoneDB 8.0): adapt testcase insert_select.test/flow_control_function.test. (#777)

### LINKED ISSUE #777: feat(StoneDB 8.0): adapt mtr insert_select.test/flow_control_function.test
**Is your feature request related to a problem? Please describe.**

1 Enable testcase ：./suite/tianmu/t/insert_select.test
    `insert...select` statement causes mysqd crashes
2 Enable mtr : ./suite/tianmu/t/flow_control_function.test
   should add cleanup: drop table;

```
INSERT INTO t1 VALUES (1,1),(2,2),(10,10);
INSERT INTO t2 (f1, f2) SELECT f1, f2 FROM t1;


### in debug version， these cetence causes that mysqld crashes
INSERT INTO t2 (f1, f2)
  SELECT f1, f1 FROM t2 src WHERE f1 < 2
  ON DUPLICATE KEY UPDATE f1 = 100 + src.f1; 
  
SELECT * FROM t2;
```
**Describe the solution you'd like**

in debug mode, it crashes
```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff6db2859 in __GI_abort () at abort.c:79
#2  0x00007ffff6db2729 in __assert_fail_base (fmt=0x7ffff6f48588 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n",
    assertion=0x55555c654b00 "!table || (!table->read_set || bitmap_is_set(table->read_set, field_index()))",
    file=0x55555c654688 "/data/codebase/stonedb80/stonedb/sql/field.cc", line=3761, function=<optimized out>) at assert.c:92
#3  0x00007ffff6dc3fd6 in __GI___assert_fail (assertion=0x55555c654b00 "!table || (!table->read_set || bitmap_is_set(table->read_set, field_index()))",
    file=0x55555c654688 "/data/codebase/stonedb80/stonedb/sql/field.cc", line=3761,
    function=0x55555c655c28 "virtual longlong Field_long::val_int() const") at assert.c:101
#4  0x0000555559615daa in Field_long::val_int (this=0x7ffab011f240) at /data/codebase/stonedb80/stonedb/sql/field.cc:3761
#5  0x000055555b4a8d27 in Tianmu::handler::(anonymous namespace)::has_dup_key (indextab=
    std::shared_ptr<class Tianmu::index::RCTableIndex> (use count 2, weak count 0) = {...}, table=0x7ffab0136fb0, row=@0x7fffe44ef0b8: 140737127852224)
    at /data/codebase/stonedb80/stonedb/storage/tianmu/handler/ha_tianmu.cpp:320
#6  0x000055555b4a93c1 in Tianmu::handler::ha_tianmu::write_row (this=0x7ffab012fdb0, buf=0x7ffab011f200 "\375\001")
    at /data/codebase/stonedb80/stonedb/storage/tianmu/handler/ha_tianmu.cpp:429
#7  0x000055555966dca4 in handler::ha_write_row (this=0x7ffab012fdb0, buf=0x7ffab011f200 "\375\001")
    at /data/codebase/stonedb80/stonedb/sql/handler.cc:7862
#8  0x00005555599dcbc7 in write_record (thd=0x7ffab0001040, table=0x7ffab0136fb0, info=0x7ffab1521080, update=0x7ffab15210f8)
    at /data/codebase/stonedb80/stonedb/sql/sql_insert.cc:1822
#9  0x00005555599de887 in Query_result_insert::send_data (this=0x7ffab1521038, thd=0x7ffab0001040, values=...)
    at /data/codebase/stonedb80/stonedb/sql/sql_insert.cc:2404
#10 0x00005555593a32c2 in Query_expression::ExecuteIteratorQuery (this=0x7ffab01604f0, thd=0x7ffab0001040)
    at /data/codebase/stonedb80/stonedb/sql/sql_union.cc:1305
#11 0x00005555593a3542 in Query_expression::execute (this=0x7ffab01604f0, thd=0x7ffab0001040) at /data/codebase/stonedb80/stonedb/sql/sql_union.cc:1343
#12 0x00005555592e79ac in Sql_cmd_dml::execute_inner (this=0x7ffab0149778, thd=0x7ffab0001040) at /data/codebase/stonedb80/stonedb/sql/sql_select.cc:798
#13 0x00005555592e6e70 in Sql_cmd_dml::execute (this=0x7ffab0149778, thd=0x7ffab0001040) at /data/codebase/stonedb80/stonedb/sql/sql_select.cc:597
#14 0x000055555925a70a in mysql_execute_command (thd=0x7ffab0001040, first_level=true) at /data/codebase/stonedb80/stonedb/sql/sql_parse.cc:3577
#15 0x000055555925ffe6 in dispatch_sql_command (thd=0x7ffab0001040, parser_state=0x7fffe44f0b90)
    at /data/codebase/stonedb80/stonedb/sql/sql_parse.cc:5239
#16 0x0000555559255a65 in dispatch_command (thd=0x7ffab0001040, com_data=0x7fffe44f14e0, command=COM_QUERY)
    at /data/codebase/stonedb80/stonedb/sql/sql_parse.cc:1959
#17 0x0000555559253adc in do_command (thd=0x7ffab0001040) at /data/codebase/stonedb80/stonedb/sql/sql_parse.cc:1362
#18 0x0000555559490415 in handle_connection (arg=0x555563ab4220)
    at /data/codebase/stonedb80/stonedb/sql/conn_handler/connection_handler_per_thread.cc:302
#19 0x000055555b3b228a in pfs_spawn_thread (arg=0x555563b3c210) at /data/codebase/stonedb80/stonedb/storage/perfschema/pfs.cc:2942
#20 0x00007ffff7fa4609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#21 0x00007ffff6eaf133 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
(gdb) f 4
#4  0x0000555559615daa in Field_long::val_int (this=0x7ffab011f240) at /data/codebase/stonedb80/stonedb/sql/field.cc:3761
3761      ASSERT_COLUMN_MARKED_FOR_READ;
(gdb) p table
$1 = (TABLE *) 0x7ffab0136fb0
(gdb) p table->read_set
$2 = (MY_BITMAP *) 0x7ffab0137108
(gdb) call bitmap_is_set(table->read_set, field_index()))
Junk after end of expression.
(gdb) call field_index()
2022-10-24T05:34:17.992723Z 0 [Note] [MY-011953] [InnoDB] Page cleaner took 69465ms to flush 0 and evict 14 pages
[Thread 0x7fffe45f5700 (LWP 2475990) exited]
$3 = 0
(gdb) cal bitmap_is_set(table->read_set,0)
$4 = false
```
**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by lujiashun**
## Summary about this PR

Issue Number: close #777

1 fix insert_select.test crash in debug mode;
  compared with stonedb5.7 flow, add prepare_for_positional_update in Query_result_insert::prepare;
2 fix "flow_control_function.test" failed in debug mode;

## Tests Check List

- [ ] Unit test
- [X] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [X] New Feature
- [ ] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Review by DandreChen — APPROVED** [review]
LGTM.

File: mysql-test/suite/tianmu/t/flow_control_function.test [hunk 1]
```diff
@@ -1,6 +1,7 @@
 --echo #
 --echo # Flow control function,Based on the test
 --echo #
```
Comment (RingsC) [inline mysql-test/suite/tianmu/t/flow_control_function.test]: Suggestion: remove these comments. It's redundent, the file name indicates what we do in this test.

Comment (RingsC) [reply mysql-test/suite/tianmu/t/flow_control_function.test]:
```
--echo #
--echo # Flow control function,Based on the test
--echo #
```

File: sql/sql_insert.cc [hunk 2]
```diff
@@ -2272,6 +2272,12 @@ bool Query_result_insert::prepare(THD *thd, const mem_root_deque<Item *> &,
 
   table = lex->insert_table_leaf->table;
 
+  // tianmu start, TODO, refer to stonedb5.7 implementation
```
Comment (RingsC) [inline sql/sql_insert.cc]: Remove the useless comment, we have git log to trance the changes.

**Review by RingsC — CHANGES_REQUESTED** [review]

**Review by RingsC — APPROVED** [review]
LGTM, Great Job

**Pull request merged**